src/components/FileTree/index.spec.tsx

```diff
@@ -1,399 +1,120 @@
-          filename: 'B',
-          path: 'parent/B',
-        }),
-        createInternalVersionEntry({
-          ...fakeVersionEntry,
-          depth: 1,
-          filename: 'A',
-          path: 'parent/A',
+      expect(dispatch).toHaveBeenCalledWith(
+        fileTreeActions.buildTree({
+          version,
         }),
-      ];
-      const version = createVersionWithEntries(entries);
-
-      const tree = buildFileTree(version);
-      const firstNode = tree.children[0] as DirectoryNode;
-
-      expect(firstNode.children).toEqual([
-        {
-          id: entries[2].path,
-          name: 'A',
-        },
-        {
-          id: entries[1].path,
-          name: 'B',
-        },
-      ]);
+      );
     });
 
-    it('puts directories first in a child node', () => {
-      const entries = [
-        createInternalVersionEntry({
-          ...fakeVersionEntry,
-          filename: 'parent',
-          mime_category: 'directory',
-          path: 'parent',
-        }),
-        createInternalVersionEntry({
-          ...fakeVersionEntry,
-          depth: 1,
-          filename: 'B',
-          mime_category: 'directory',
-          path: 'parent/B',
-        }),
-        createInternalVersionEntry({
-          ...fakeVersionEntry,
-          depth: 1,
-          filename: 'A',
-          path: 'parent/A',
-        }),
-      ];
-      const version = createVersionWithEntries(entries);
-
-      const tree = buildFileTree(version);
-      const firstNode = tree.children[0] as DirectoryNode;
-
-      expect(firstNode.children).toEqual([
-        {
-          id: entries[1].path,
-          name: 'B',
-          children: [],
-        },
-        {
-          id: entries[2].path,
-          name: 'A',
-        },
-      ]);
-    });
-  });
+    it('does not dispatch buildTree when tree is defined', () => {
+      const store = configureStore();
+      const version = getVersion({ store });
+      _buildTree(store, version);
 
-  describe('FileTree', () => {
-    const getVersion = ({
-      store = configureStore(),
-      version = fakeVersion,
-    }) => {
-      store.dispatch(versionActions.loadVersionInfo({ version }));
+      const dispatch = spyOn(store, 'dispatch');
 
-      return getVersionInfo(store.getState().versions, version.id);
-    };
+      render({ store, versionId: version.id });
 
-    const render = ({
-      _log = createFakeLogger(),
-      onSelect = jest.fn(),
-      store = configureStore(),
-      versionId = 1234,
-    } = {}) => {
-      return shallowUntilTarget(
-        <FileTree _log={_log} versionId={versionId} onSelect={onSelect} />,
-        FileTreeBase,
-        {
-          shallowOptions: { context: { store } },
-        },
-      );
-    };
+      expect(dispatch).not.toHaveBeenCalled();
+    });
 
     it('renders a ListGroup component with a Treefold', () => {
       const store = configureStore();
       const version = getVersion({ store });
+      _buildTree(store, version);
```

Comment:
Could this be moved in `render()`? It looks like it is used everywhere.

Reply:
There are a couple of issues with that, which is why I didn't do it.

1. We don't always want to build the file tree, so I would have needed an argument like `buildFileTree: boolean`.
2. The `render` function is designed to accept a `versionId`, not a `version`. It's up to the test to decide whether to load a `version` that matches the `versionId` into the store before calling `render`. `buildTree` requires an actual `version`, so it would require changes to the `render` function, which is currently working pretty well for the tests we need.

I did experiment with this, but decided that keeping `render` the way it is would be best. Suggestions for improvements are welcome, though.